src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -111,7 +111,10 @@ public override void VisitRazorDirective(RazorDirectiveSyntax node)
 
         public override void VisitRazorDirectiveBody(RazorDirectiveBodySyntax node)
         {
-            AddSemanticRange(node.Keyword, SyntaxKind.RazorDirective);
+            if (node.Keyword.Kind != SyntaxKind.CSharpStatementLiteral)
```

Comment:
Hmm, is there a way to restrict this more specifically to `using` via some keyword check?

Reply:
The "keyword" in this case is actually `using Some.Namespace`. Further, we need to not colorize anything that CSharp might colorize, because once my PR goes in that could lead to overlap between results, which the spec strictly forbids.